buildSrc/src/main/groovy/io.deephaven.java-toolchain-conventions.gradle

```diff
@@ -116,12 +116,24 @@ def compilerArgs = { String compilerDirectivesFile ->
   ]
 }
 
-// Utility to add jvm args to all executions, whether intellij or from a application script or gradle javaexec
-def devJvmArgs = [
-//  '-XX:Tier4CompileThreshold=1000', // this optional line makes it easier to trigger the c2 error on the above methods
-//  '-XX:+PrintFlagsFinal',           // this optional line makes it easier to explore the final values for compiler args
-//  '-XX:+PrintCompilation',          // this optional line shows jit operations as they happen
-]
+def parseJvmArgumentsFromEnv = { String envName, List<String> orElse ->
+  def s = System.getenv(envName)
+  if (s == null) {
+    return orElse
+  }
+  // Note: this logic could be improved to handle quotes and other cases, but hopefully this suffices for now.
```

Comment:
So if you provide args, instead of adding to the JVM args it just replaces all of them - is that what we want?

Reply:
I'm not sure I understand the question. The limitation right now is that "complex" cases aren't handled. Something like

```shell
START_OPTS="-Dmy.prop=\"my value with spaces\"" ...
```

is not currently parsed correctly.

Reply:
There's default JVM Args:
```
  '-XX:+UseG1GC',
  '-XX:MaxGCPauseMillis=100',
  '-XX:+UseStringDeduplication',
```
Now if you specify `JAVA_OPTS="-XX:+PrintGCDetails"`, you'll lose those default JVM args instead of it being added to those args. Just wondering if that's the behaviour we want - I would've expected JVM args I provide to be in addition to the default args (I think)

Reply:
Yep, this is the behavior we want IMO.

https://deephaven.io/core/docs/how-to-guides/configuration/native-application/#native-application-script

> JAVA_OPTS is an environment variable that may contain pre-defined Deephaven recommended arguments. The user is able to set their own JAVA_OPTS environment variable to override the Deephaven defaults.

If you want to "append", then START_OPTS is the way to go, and that will be ok for `START_OPTS="-XX:+PrintGCDetails"`

The reason you might want to override JAVA_OPTS is to set another GC (which won't work w/ START_OPTS b/c it conflicts w/ JAVA_OPTS): `JAVA_OPTS="-XX:+UseZGC"`.